app/src/main/java/com/ncnf/user/FriendsTrackerActivity.java

```diff
@@ -0,0 +1,318 @@
+
+    // for testing
+    //private static final String uuid = "MSpKLkyyrrN3PC5KmxkoD05Vy1m2";
+    //private static final String uuid2 = "xsohP7PYdDQZ69STCpznOZt0Zfg2";
+
+    private Marker marker;
+
+    //private DatabaseService dbs;
+
+    private LatLngBounds bounds;
+
+    @SuppressLint("UseCompatLoadingForDrawables")
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_friends_tracker);
+
+        //dbs = new DatabaseService();
+
+        //dbs.updateField(USERS_COLLECTION_KEY + uuid, USER_LOCATION_KEY, new GeoPoint(46.5201852, 6.5637122));
+        //dbs.updateField(USERS_COLLECTION_KEY + uuid2, USER_LOCATION_KEY, new GeoPoint(46.516981, 6.57144331));
+
+        counter = 0;
+
+        friendsUUID = new ArrayList<>();
+        markers = new ArrayList<>();
+
+        //friendsUUID.add(uuid);
+        //friendsUUID.add(uuid2);
+
+        findUserButton = findViewById(R.id.find_user_button);
+
+        //user = CurrentUserModule.getCurrentUser();
+        user.loadUserFromDB();
+
+        myFusedLocationClient = LocationServices.getFusedLocationProviderClient(this);
+
+        mapView = findViewById(R.id.friends_map);
+        mapView.onCreate(savedInstanceState);
+        mapView.getMapAsync(googleMap -> {
+            mMap = googleMap;
+            onMapReady(mMap);
+            getLastKnownLocation();
+
+            //changeUserLocs();
+
+            findUserButton.setOnClickListener(v -> {
+                if(user.getLoc() != null) {
+                    setMapCamera(user.getLoc());
+                }
+            });
+        });
+
+
+    }
+
+    private void startLocationUpdates() {
+        handler.postDelayed(runnable = new Runnable() {
+            @Override
+            public void run() {
+                getUserLocations();
+                handler.postDelayed(runnable, LOCATION_UPDATE_INTERVAL);
+            }
+        }, LOCATION_UPDATE_INTERVAL);
+    }
+
+    // ONLY FOR TESTING / DEMOS
+    /**
+    private void changeUserLocs() {
+        handler2.postDelayed(runnable2 = new Runnable() {
+            @Override
+            public void run() {
+                dbs.updateField(USERS_COLLECTION_KEY + uuid, USER_LOCATION_KEY, new GeoPoint(46.51612823811807 + counter*0.0005, 6.560711384991796)).thenAccept(aBoolean -> {
+                    dbs.updateField(USERS_COLLECTION_KEY + uuid2, USER_LOCATION_KEY, new GeoPoint(46.51612823811807, 6.560711384991796+ counter*0.0005)).thenAccept(aBoolean1 -> {
+                        counter += 1;
+                    });
+                });
+                handler2.postDelayed(runnable2, 2*LOCATION_UPDATE_INTERVAL);
+            }
+        }, 2*LOCATION_UPDATE_INTERVAL);
+    }
+     **/
+
+    private void stopLocationUpdates(){
+        handler.removeCallbacks(runnable);
+        //handler2.removeCallbacks(runnable2);
+    }
+
+    private void getUserLocations() {
```

Comment:
Well done for this one! It must have been quite tricky to implement.
It also helped me implement the map db, so thanks!

Reply:
Yep, well done !